internal/api/handlers.go

```diff
@@ -0,0 +1,150 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/Azzonya/go-shortener/internal/entities"
+	"github.com/gin-gonic/gin"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type Request struct {
+	URL string `json:"url"`
+}
+
+type Response struct {
+	Result string `json:"result"`
+}
+
+func (o *Rest) ShortenJSON(c *gin.Context) {
+	var err error
+	var exist bool
+
+	req := &Request{}
+	resp := Response{}
+
+	err = c.BindJSON(req)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Failed to read body",
+			"error":   err.Error(),
+		})
+		return
+	}
+
+	outputURL, err := o.shortener.ShortenAndSaveLink(req.URL)
```

Comment:
```suggestion
	resp.Result, err = o.shortener.ShortenAndSaveLink(req.URL)
```
а можно сразу записать в нужное место и не выделять память под лишнюю переменную (в целом это важно только если объем ответа большой, но в данном случае еще и сокращается количество кода